core/graph/src/test/java/uk/gov/gchq/gaffer/graph/GraphTest.java

```diff
@@ -2418,20 +1970,94 @@ public void shouldNotAddDefaultSecurityHooksIfAlreadyAdded() {
         TestStore.mockStore = mock(Store.class);
         given(TestStore.mockStore.isSupported(NamedOperation.class)).willReturn(true);
         // When
-        final GraphConfig config = new GraphConfig.Builder()
-                .graphId("test")
-                .addHook(new FunctionAuthoriser(Lists.newArrayList(Identity.class)))
+        final GraphConfig config = new GraphConfig.Builder().graphId("test")
+                .addHook(new FunctionAuthoriser(Lists.newArrayList(Identity.class))).build();
+
+        final Graph graph = new Graph.Builder().addSchemas(StreamUtil.schemas(getClass()))
+                .storeProperties(storeProperties).config(config).build();
+
+        // Then
+        assertEquals(Arrays.asList(NamedOperationResolver.class, NamedViewResolver.class, FunctionAuthoriser.class),
+                graph.getGraphHooks());
+        assertEquals(Identity.class,
+                ((FunctionAuthoriser) graph.getConfig().getHooks().get(2)).getUnauthorisedFunctions().get(0));
+    }
+
+    @Test
+    public void shouldExpandGlobalEdges() {
```

Comment:
Might be worth adding some integration tests into the federated store to check the behaviour